### DISCUSSION TITLE
Is there a way to cut out part of the image?

### DISCUSSION THREAD

**Discussion by loaa4** [PMEAL/porespy#695] [Q&A] [answered]
Dear professor,
I use [this function](https://github.com/PMEAL/porespy/discussions/680) to extract network,but the file I use(It has nearly 50,000 particles) was too big to extract one time,so I clip the file into 64 parts and extract them separately,but in this circumstance,the boundary of the figure will have more void cause by missing particles.
So I wonder is there a function that can cut out part of the boundary image,so these void won't influence the extract of pores.
Any suggestion will be helpful.
![WM7{ O)ZWT70AV FN5TK88N](https://user-images.githubusercontent.com/99061790/169047416-ce466c26-e967-45ab-b2af-4eb16af0a543.png)

**Comment by jgostick** [comment, marked as answer]
If I understand you correctly, you're generating 64 subimages by scanning the csv file, and putting each sphere into it's respective image, but the problem arises because spheres on the borders span between multiple images.  This is conceptually simple to solve, but practially tricky.  The solution is the generate sub-images that are larger than you need, by the R of your largest expected particle.  These border spheres then end up in multiple images.  Finally you just crop the images down to the correct size so there is no overlap, and your done. 

In practice howevrer this padding will create lots of complications with indexing and cropping, etc.  It's possible and porespy has a few tools that may help you.  In the tools module we have ``subdivide`` and ``recombine``, which are meant for applying filters on sub-images then joining the result.  I suggest you take a look at those to see if they can help you at all.  Other than than, we do not have a function that does exactly what you're looking for.
